tools/cli.js

```diff
@@ -116,9 +124,41 @@ commandProcessor.complete = (inputs, depth) => {
 
 commandProcessor.call = (interfaceName, methodName, args, callback) => {
   if (!state.client) return callback(new Error('Not connected'));
+
+  interfaceName = tryComplete(interfaceName, Object.keys(state.api));
+  methodName = tryComplete(methodName, state.api[interfaceName]);
+
   state.connection.callMethod(interfaceName, methodName, args, callback);
 };
 
+commandProcessor.call.complete = (inputs, depth) => {
+  if (!state.api) return [[], depth];
+
+  const iface = inputs[depth++];
+  let method = inputs[depth];
+  if (method === '' && inputs[depth + 1] !== undefined) {
+    method = inputs[++depth];
+  }
+
+    // (a and b and c) or (!a and (b or c))
+  let completions = complete(iface, Object.keys(state.api));
+  if (completions.length !== 1 && method === undefined ||
```

Comment:
Can we have a comment explaining the logic behind these conditions above the `if` statement? Aside, how about slitting it into three `if`s with a duplicate body (it's short enough to do that). I'd find it more readable.